### PR TITLE
Hotfix/adding api

### PR DESCRIPTION
**Pull request by marla-singer**
### Changes
1. addapibackendutils.js
   - Rewrite feature of adding api
2. addapibackend.js
   - Rewrite condition for require fields, for unique field
   - Fix "after hook"

WIP

### CODE REVIEW

**Comment by jykae** [comment]
@marla-singer if you are able to figure out how to fix my PRs "element not visible" issue for modal, I would be grateful. Reviewing this one.

**Comment by jykae** [comment]
Cool, ran tests against nightly, just assertion in last on fails.
nayttokuva 2016-10-20 kello 12 06 54

File: test/common.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var webdriver = require('selenium-webdriver');
 var By = require('selenium-webdriver').By;
 
 var CommonUtils = {
-    url: 'https://nightly.apinf.io',
+    url: 'http://localhost:3030/',
```
Comment (jykae) [inline test/common.js]: I would keep this as nightly like it was, could you change it back.

Comment (marla-singer) [reply test/common.js]: Oh, yeah! I'm changing it back to nightly, of course

File: test/addapibackend.js [hunk 2]
```diff
@@ -92,16 +113,31 @@ test.describe('Add API Backend', function() {
...
+        var errorElement = driver.findElement(By.xpath('//*[@id="addApiForm"]/fieldset/div[3]/span'));
+        errorElement.getText().then(function(text){
+            assert.include(text, 'URL must be a valid URL');
         });
-        var errorElement = driver.findElement(By.xpath('//*[@class="s-alert-box-inner"]/p'));
+    });
+    test.it('8.5 should not add/publish api with no unique name', function() {
+        newUser.added = false;
+        CommonUtils.signIn(driver);
+        CommonUtils.fillSignInForm(driver, 'test@test.test', 'testuser');
```
Comment (jykae) [inline test/addapibackend.js]: I created user 'robot@test.com', 'robottest' on nightly to run tests, maybe we could use that, could you change these accordingly?

Comment (marla-singer) [reply test/addapibackend.js]: I will change

**Comment by marla-singer** [comment]
@jykae '8.5' test works on localhost and doesn't correct work on nightly :confused:
It returned an empty line instead of 'API Name must be unique'. 
Probably our problems are the same

**Comment by jykae** [comment]
@marla-singer Hmh, I ran just that 5 times in a row for nightly, 2/5 success. It must be some sort of race condition issue, text not available yet. If it would be somehow possible to wait a moment?

**Comment by marla-singer** [comment]
@jykae Waiting a moment is a good solution, I thought about this too and selenium web-driver provides this feature

``` javascript
driver.wait(function() {
      ...
         }, timeout);
```

Hovewer, I haven't thought up yet how I can use it

File: test/addapibackend.js [hunk 3]
```diff
@@ -32,76 +37,107 @@ test.describe('Add API Backend', function() {
...
+        CommonUtils.signIn(driver);
+        CommonUtils.fillSignInForm(driver, 'robot@test.com', 'robottest');
+        CommonUtils.goToDashboard(driver);
+        var values = {
+            apiName: 'NewBackend',
+            apiDescription: 'Test Description',
+            apiURL: 'http://google.com'
+        };
+        AddAPIBackendUtil.addNewBackend(driver, values);
+        // Get the API Name field in form
```
Comment (jykae) [inline test/addapibackend.js:139]: @marla-singer Tried line driver.sleep(1000); after adding.
and test passes better 5/5 runs for nightly.

**Comment by jykae** [comment]
@marla-singer Great I think this is ready to merge!

**Pull request merged**